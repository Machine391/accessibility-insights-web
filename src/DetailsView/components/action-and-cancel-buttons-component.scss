// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

@import '../../common/styles/common.scss';

.action-and-cancel-buttons-component {
    display: flex;
    justify-content: flex-end;

    .action-cancel-button-col + .action-cancel-button-col {
        margin-left: 8px;
    }

    .cancel-button {
        font-size: 14px;
        font-weight: 600;
        height: 32px;
        border: 1px solid $neutral-60;
        color: $neutral-60;
        border-radius: 2px;

        &:hover {
            border: 1px solid $neutral-60;
            font-weight: 600;
            color: $neutral-70;
            background-color: $neutral-6-5;
            border-radius: 2px;
        }
    }

    .action-button-disabled {
        font-size: 14px;
        font-weight: 600;
        height: 32px;
        border: 1px solid $neutral-4;
        background-color: $neutral-4;
        color: $neutral-10;
        pointer-events: none;
        border-radius: 2px;
        border-width: 10px;
    }

    .action-button-enabled {
        font-size: 14px;
        font-weight: 600;
        height: 32px;
        border: 1px solid $communication-primary;
        background-color: $communication-primary;
        color: $always-white;
        border-radius: 2px;
        border-width: 10px;

        &:hover {
            background-color: $communication-primary;
            color: $always-white;
            border-radius: 2px;
            border-width: 10px;
        }
    }
}
