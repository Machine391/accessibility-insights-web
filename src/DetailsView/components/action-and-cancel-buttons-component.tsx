// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { Button } from '@fluentui/react-components';
import { isEmpty } from 'lodash';
import * as React from 'react';
import styles from './action-and-cancel-buttons-component.scss';

export interface ActionAndCancelButtonsComponentProps {
    isHidden: boolean;
    primaryButtonDisabled: boolean;
    primaryButtonText: string;
    primaryButtonOnClick: (ev) => void;
    cancelButtonOnClick: (ev) => void;
    primaryButtonHref?: string;
    primaryButtonDataAutomationId?: string;
}

export class ActionAndCancelButtonsComponent extends React.Component<ActionAndCancelButtonsComponentProps> {
    public render(): JSX.Element {
        return (
            <div className={styles.actionAndCancelButtonsComponent} hidden={this.props.isHidden}>
                <div className={styles.actionCancelButtonCol}>
                    <Button
                        className={styles.cancelButton}
                        onClick={this.props.cancelButtonOnClick}
                    >
                        Cancel
                    </Button>
                </div>
                <div className={styles.actionCancelButtonCol}>
                    <Button
                        appearance="primary"
                        className={
                            this.props.primaryButtonDisabled
                                ? styles.actionButtonDisabled
                                : styles.actionButtonEnabled
                        }
                        data-automation-id={this.props.primaryButtonDataAutomationId}
                        onClick={this.props.primaryButtonOnClick}
                        disabled={this.props.primaryButtonDisabled}
                        href={this.props.primaryButtonHref}
                        target={isEmpty(this.props.primaryButtonHref) ? '_self' : '_blank'}
                    >
                        {' '}
                        {this.props.primaryButtonText}{' '}
                    </Button>
                </div>
            </div>
        );
    }
}
