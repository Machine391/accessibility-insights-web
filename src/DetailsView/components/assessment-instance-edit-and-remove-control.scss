// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import '../../common/styles/common.scss';

.remove-button {
    margin-left: 12px;
    font-size: 16px;
    line-height: 24px;
    color: $negative-outcome !important;
    margin-left: 12px !important;
    font-size: 16px !important;

    &:hover,
    &:focus {
        color: linktext !important;
    }

    @media screen and (forced-colors: active) {
        @include fluent-forced-color-adjust-override;

        color: inherit !important;

        &:hover,
        &:focus {
            color: initial !important;
        }
    }
}
