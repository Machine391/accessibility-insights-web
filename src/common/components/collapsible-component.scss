// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

@import '../../common/styles/common.scss';

span.collapsible-title {
    padding-left: 0.5vw;
}

.collapsible {
    display: inline-flex;
    font-size: 26px;
    height: auto;
    width: auto;
    text-align: left;
    color: #106ebe;

    h2 {
        font-weight: normal;
        font-size: 17px;
    }
}

.collapsible-content {
    overflow: hidden;
    padding-left: calc(0.5vw + 13px); // 13px is Chevron width.
}
