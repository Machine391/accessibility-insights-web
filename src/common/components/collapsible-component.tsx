// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { ActionButton } from '@fluentui/react';
import { Button, themeToTokensObject, webLightTheme } from '@fluentui/react-components';
import { css } from '@fluentui/utilities';
import styles from 'common/components/collapsible-component.scss';
import * as React from 'react';
import { ChevronDown24Regular, ChevronRight24Regular } from '@fluentui/react-icons';

export interface CollapsibleComponentProps {
    header: JSX.Element;
    content: JSX.Element;
    contentClassName?: string;
    containerClassName?: string;
}

interface CollapsibleComponentState {
    showContent: boolean;
}

export class CollapsibleComponent extends React.Component<
    CollapsibleComponentProps,
    CollapsibleComponentState
> {
    private readonly iconNameDown = 'ChevronDown';
    private readonly iconNameUp = 'ChevronRight';

    constructor(props: CollapsibleComponentProps) {
        super(props);
        this.state = { showContent: true };
    }

    private onClick = (): void => {
        this.setState(prevState => ({ showContent: !prevState.showContent }));
    };

    public render(): JSX.Element {
        const showContent = this.state.showContent;
        let iconName = this.iconNameUp;
        let content: JSX.Element | null = null;

        if (showContent) {
            iconName = this.iconNameDown;
            content = (
                <div className={css(this.props.contentClassName, styles.collapsibleContent)}>
                    {this.props.content}
                </div>
            );
        }

        return (
            <div className={css(this.props.containerClassName)}>
                {/* 111<ActionButton
                    className={styles.collapsible}
                    iconProps={{ iconName: iconName }}
                    onClick={this.onClick}
                    aria-expanded={showContent}
                >
                    {this.props.header}
                </ActionButton> */}
                <Button className={styles.collapsible} appearance='transparent'
                    onClick={this.onClick}
                    aria-expanded={showContent}>
                    {showContent ? (
                        <ChevronDown24Regular color={themeToTokensObject(webLightTheme).colorCompoundBrandStrokeHover} className={styles.collapsible} />
                    ) : (
                        <ChevronRight24Regular color={themeToTokensObject(webLightTheme).colorCompoundBrandStrokeHover} className={styles.collapsible} />
                    )}
                    111
                    {this.props.header}
                </Button>
                {content}
            </div>
        );
    }
}
