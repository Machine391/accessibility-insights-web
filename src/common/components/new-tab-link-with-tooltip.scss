// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import '../../common/styles/colors.scss';

:global(.high-contrast-theme) {
    .insights-tooltip {
        outline-style: solid;
        outline-width: 1px;
        outline-color: $always-white;
    }
}

.insights-tooltip-host {
    display: inline-block;
    min-width: 0;
}

.insights-link {
    display: flex;

    &:hover,
    &:active {
        text-decoration: none !important;
    }

    &:focus-visible {
        box-shadow: $neutral-60 0 0 0 1px inset !important;
        outline: $neutral-60 auto 1px !important;
        text-decoration: none !important;
    }
}
