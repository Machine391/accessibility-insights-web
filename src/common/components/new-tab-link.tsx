// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { Link, LinkProps, mergeClasses } from '@fluentui/react-components';
import * as React from 'react';

import { NamedFC } from '../react/named-fc';

export type NewTabLinkProps = LinkProps & { as?: 'a' };

export const NewTabLink = NamedFC<NewTabLinkProps>('NewTabLink', ({ className, ...props }) => {
    return <Link className={mergeClasses('insights-link', className)} target="_blank" {...props} />;
});
