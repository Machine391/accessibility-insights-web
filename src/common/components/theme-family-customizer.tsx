// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { ThemeProvider, PartialTheme } from '@fluentui/react';
import { FluentProvider, webDarkTheme, webLightTheme } from '@fluentui/react-components';
import { NamedFC } from 'common/react/named-fc';
import { DefaultTheme } from 'common/styles/default-theme';
import { FastPassTheme } from 'common/styles/fast-pass-theme';
import { FastPassV9Theme } from 'common/styles/fast-pass-v9-theme';
import { HighContrastTheme } from 'common/styles/high-contrast-theme';
import { UserConfigurationStoreData } from 'common/types/store-data/user-configuration-store';
import * as React from 'react';

export type ThemeFamily = 'default' | 'fast-pass';

const themeFamilyDefaultThemes: { [themeFamily in ThemeFamily]: PartialTheme } = {
    default: DefaultTheme,
    'fast-pass': FastPassTheme,
};

const themeFamilyDefaultV9Themes: { [themeFamily in ThemeFamily] } = {
    default: webLightTheme,
    'fast-pass': FastPassV9Theme,
};

export type ThemeFamilyCustomizerProps = {
    userConfigurationStoreData: UserConfigurationStoreData;
    themeFamily: ThemeFamily;
    children: React.ReactNode;
};

// This is a store-state-aware version of the Fluent UI <ThemeProvider> component. It's intended
// as a wrapper for Fluent UI-based components that need to use a different Theme from our default
// one, but still want to support being overriden by HighContrastTheme when a user has enabled our
// High Contrast Mode setting (or a native system High Contrast setting).
export const ThemeFamilyCustomizer = NamedFC<ThemeFamilyCustomizerProps>(
    'ThemeCustomizer',
    props => {
        const isHighContrastEnabled = props.userConfigurationStoreData?.enableHighContrast === true;
        const defaultTheme = themeFamilyDefaultThemes[props.themeFamily];
        const defaultV9Theme = themeFamilyDefaultV9Themes[props.themeFamily]
        const activeTheme = isHighContrastEnabled ? HighContrastTheme : defaultTheme;
        const activeV9Theme = isHighContrastEnabled ? { ...webDarkTheme, colorNeutralBackground1: '#161616 ' } : defaultV9Theme;
        return <ThemeProvider applyTo="body" theme={activeTheme}><FluentProvider theme={activeV9Theme}>{props.children}</FluentProvider></ThemeProvider>;
    },
);
