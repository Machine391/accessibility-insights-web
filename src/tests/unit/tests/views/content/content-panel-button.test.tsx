// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { Button, themeToTokensObject, webLightTheme } from '@fluentui/react-components';
import { render } from '@testing-library/react';
import * as React from 'react';
import {
    expectMockedComponentPropsToMatchSnapshots,
    mockReactComponents,
} from 'tests/unit/mock-helpers/mock-module-helpers';
import { ContentPage } from 'views/content/content-page';
import { ContentPanelButton } from 'views/content/content-panel-button';
import { ContentActionMessageCreator } from '../../../../../common/message-creators/content-action-message-creator';
import { Theme, ThemeDeps, ThemeInnerProps } from 'common/components/theme';
import { Mock } from 'typemoq';
import { DocumentManipulator } from 'common/document-manipulator';

jest.mock('@fluentui/react-components');

describe('ContentPanelButton', () => {
    mockReactComponents([Button]);
    const documentManipulatorMock = Mock.ofType(DocumentManipulator);
    let props: ThemeInnerProps;
    beforeEach(() => {
        props = {
            deps: {
                documentManipulator: documentManipulatorMock.object,
                storeActionMessageCreator: null,
                storesHub: null,
            } as ThemeDeps,
            storeState: {
                userConfigurationStoreData: {
                    enableHighContrast: null,
                },
            },
        } as ThemeInnerProps;
    });

    const content = {
        for: {
            testing: ContentPage.create(() => 'CONTENT FOR TESTING' as any),
        },
    };
    const contentTitle = 'TITLE FOR TESTING';

    const deps = {
        contentProvider: ContentPage.provider(content),
        contentActionMessageCreator: {} as any as ContentActionMessageCreator,
    };



    it('renders from content', () => {
        const renderResult = render(
            // <Theme deps={props.deps}>
            <ContentPanelButton
                deps={deps}
                reference={content.for.testing}
                iconName="info"
                contentTitle={contentTitle}
            >
                TEXT
            </ContentPanelButton>,
            // </Theme>
        );

        expect(renderResult.asFragment()).toMatchSnapshot();
        expectMockedComponentPropsToMatchSnapshots([Button]);
    });

    it('renders from path', () => {
        const renderResult = render(
            <ContentPanelButton
                deps={deps}
                reference={'for/testing'}
                iconName="iconName"
                contentTitle={contentTitle}
            >
                TEXT
            </ContentPanelButton>,
        );

        expect(renderResult.asFragment()).toMatchSnapshot();
        expectMockedComponentPropsToMatchSnapshots([Button]);
    });

    it('renders without iconName', () => {
        const renderResult = render(
            <ContentPanelButton deps={deps} reference={'for/testing'} contentTitle={contentTitle}>
                TEXT
            </ContentPanelButton>,
        );

        expect(renderResult.asFragment()).toMatchSnapshot();
        expectMockedComponentPropsToMatchSnapshots([Button]);
    });
});
