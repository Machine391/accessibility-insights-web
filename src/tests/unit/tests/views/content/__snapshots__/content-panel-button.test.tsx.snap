// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentPanelButton renders from content 1`] = `
<DocumentFragment>
  <mock-button
    appearance="transparent"
    aria-label="info and examples"
    classname="info"
    icon="[object Object]"
  >
    TEXT
  </mock-button>
</DocumentFragment>
`;

exports[`ContentPanelButton renders from content: Button props 1`] = `
{
  "appearance": "transparent",
  "aria-label": "info and examples",
  "children": "TEXT",
  "className": "info",
  "icon": <InfoRegular />,
  "onClick": [Function],
}
`;

exports[`ContentPanelButton renders from path 1`] = `
<DocumentFragment>
  <mock-button
    appearance="transparent"
    aria-label="info and examples"
    classname="info"
  >
    TEXT
  </mock-button>
</DocumentFragment>
`;

exports[`ContentPanelButton renders from path: Button props 1`] = `
{
  "appearance": "transparent",
  "aria-label": "info and examples",
  "children": "TEXT",
  "className": "info",
  "icon": undefined,
  "onClick": [Function],
}
`;

exports[`ContentPanelButton renders without iconName 1`] = `
<DocumentFragment>
  <mock-button
    appearance="transparent"
    aria-label="info and examples"
    classname="info"
  >
    TEXT
  </mock-button>
</DocumentFragment>
`;

exports[`ContentPanelButton renders without iconName: Button props 1`] = `
{
  "appearance": "transparent",
  "aria-label": "info and examples",
  "children": "TEXT",
  "className": "info",
  "icon": undefined,
  "onClick": [Function],
}
`;
